extensions/azure/cosmos/contract-definition-store-cosmos/src/main/java/org/eclipse/dataspaceconnector/contract/definition/store/CosmosContractDefinitionStore.java

```diff
@@ -125,8 +125,9 @@ public void update(ContractDefinition definition) {
     }
 
     @Override
-    public void delete(String id) {
-        cosmosDbApi.deleteItem(id);
+    public ContractDefinition deleteById(String id) {
+        var deletedItem = cosmosDbApi.deleteItem(id);
+        return deletedItem == null ? null : convert(deletedItem);
```

Comment:
Question: Do we want to return `null` here when the contract is not found or do want want to throw an error indicating that it was not found?  Either way is fine but want to be clear on the intention

Reply:
I think this is the expected behaviour because it was the case for story #801, so for consistency I would do it like this.

Reply:
I confirmed with Paul in issue 802 that this is the expected behaviour.